src/mongoose_acc.erl

```diff
@@ -121,40 +129,57 @@ new(#{ location := Location, lserver := LServer } = Params) ->
       non_strippable => sets:new()
      }.
 
+-spec ref(Acc :: t()) -> reference().
 ref(#{ mongoose_acc := true, ref := Ref }) ->
     Ref.
 
+-spec timestamp(Acc :: t()) -> erlang:timestamp().
 timestamp(#{ mongoose_acc := true, timestamp := TS }) ->
     TS.
 
+-spec lserver(Acc :: t()) -> jid:lserver().
 lserver(#{ mongoose_acc := true, lserver := LServer }) ->
     LServer.
 
+-spec element(Acc :: t()) -> exml:element() | undefined.
 element(#{ mongoose_acc := true, stanza := #{ element := El } }) ->
     El;
 element(#{ mongoose_acc := true }) ->
     undefined.
 
+-spec from_jid(Acc :: t()) -> jid:jid() | undefined.
 from_jid(#{ mongoose_acc := true, stanza := #{ from_jid := FromJID } }) ->
     FromJID;
 from_jid(#{ mongoose_acc := true }) ->
     undefined.
 
+-spec to_jid(Acc :: t()) -> jid:jid() | undefined.
 to_jid(#{ mongoose_acc := true, stanza := #{ to_jid := ToJID } }) ->
     ToJID;
 to_jid(#{ mongoose_acc := true }) ->
     undefined.
 
+-spec packet(Acc :: t()) -> ejabberd_c2s:packet() | undefined.
```

Comment:
it's just a type declaration, don't see a big difference where to declare it.
it can be avoided, but i don't see any reason to make 3 calls instead of one.
also, it's more routing semantics than C2S.